19 criar coleção de defensores

Murilo-Sanches opened the pull request:
Closes #19

Bestiário.

Murilo-Sanches committed:
feat: um bestiário extremamente simples

A tela é dividida em duas partes por uma linha vertical, a esquerda se
encontra os botões das criaturas e ao serem clicadas, o conteúdo da
direita é atualizado com o nome, descrição e imagem do mob.

Murilo-Sanches committed:
feat: componentizando o botão e permitindo callbacks quando forem clicados

Murilo-Sanches committed:
feat: centralizando o uso de cores e manager para trackear o mouse

Murilo-Sanches committed:
fix: as cores não estavam renderizando

Agora as cores são consumidas de uma posição fixa na memória.

Murilo-Sanches committed:
refactor: usando o componente button

Murilo-Sanches committed:
fix: faltou a assinatura da função

Murilo-Sanches committed:
fix: garantindo que o evento de click persista e seja acessível no mesmo frame que aconteceu

Murilo-Sanches committed:
refactor: apontando para os headers depois da mudança no makefile

Murilo-Sanches committed:
feat: simulando um callback de verdade, onde é possível passar dados para o on_click

Murilo-Sanches committed:
refactor: removendo textos longos; removendo/deixando conteúdo desnecessário inline

Murilo-Sanches committed:
feat: renderizando defensores e inimigos

Murilo-Sanches committed:
refactor: não há necessidade de diferenciar entre defensor e inimigo

Murilo-Sanches committed:
style

Murilo-Sanches committed:
fix: resolvendo alguns warnings do compilador

Murilo-Sanches committed:
fix: deixando a visibilidade da tela atual pública

Murilo-Sanches committed:
feat: botão de voltar

Murilo-Sanches committed:
Merge branch 'develop' of github.com:guardioes-da-imunidade/guardioes-da-imunidade into 19-criar-coleção-de-defensores

Murilo-Sanches committed:
feat: voltando para o lobby e indo para o bestiário

Murilo-Sanches committed:
fix: ajustando imports

Murilo-Sanches committed:
style

Murilo-Sanches committed:
fix: tratando eventos de resize da janela

Murilo-Sanches committed:
feat: centralizando o uso de fontes

Murilo-Sanches committed:
fix: considerando a outra possibilitade para click

Murilo-Sanches committed:
refactor

Murilo-Sanches committed:
style

Murilo-Sanches committed:
feat: bestiário pronto, falta apenas o background custom

Murilo-Sanches committed:
fix: ajustando posições e a imagem correta

Murilo-Sanches committed:
Merge branch 'develop' of https://github.com/guardioes-da-imunidade/guardioes-da-imunidade into 19-criar-coleção-de-defensores

Murilo-Sanches committed:
fix: para não conflitar com as structs  já existentes

Murilo-Sanches committed:
fix: ajustando posicionamento dos elementos

Murilo-Sanches committed:
refactor: organizando imports

paulorccamargo commented on src/core/palette.h:
Retire os comentários, não é necessário pois os nomes das váriaveis globais estão bem descritivos.

paulorccamargo commented on src/entities/entities.c:
Um dos objetivos principais da funcionalidade da coleção é descrever de forma clara para o jogador o que aquelas representações de personagens significam na vida real, sendo assim, é necessário uma descrição científica para que durante o jogo, o jogador aprenda sobre o tema do projeto, sistema imunológico e seus agentes.

paulorccamargo commented on src/screens/bestiary/bestiary.c:
Melhorando a descrição de forma educativa e científica para o jogador como já dito em comentários anteriores, coloque na tela também os atributos que você já tem de cada agente, assim, deixando para o jogador tomar a própria decisão em escolher os personagens para sua coleção removendo a necessidade de "Defensor geral" por exemplo, assim deixando mais claro a presença que aquele personagem tem em campo.

paulorccamargo reviewed (changes requested):
Solicitações importantes para manter o objetivo do projeto, principalmente para o tema educacional.

Murilo-Sanches committed:
chore: removendo comentários

Murilo-Sanches committed:
style: adicionei infos científicas nas entidades

Murilo-Sanches committed:
feat: mostrando o status das entidades

Murilo-Sanches committed:
refactor(wip): centralizando a estrutura dos inimigos no arquivo entities

Murilo-Sanches committed:
fix: corrigindo para o tipo certo

Murilo-Sanches committed:
fix: inicializando sem lixo de memória

Murilo-Sanches committed:
fix: inicializando sem lixo de memória

Murilo-Sanches committed:
feat: permitindo que além de funcionar como bestiário, funcione como loja para equipar os defensores

Murilo-Sanches committed:
refactor(wip): mudando os códigos hard-coded para apontar para a struct centralizada

paulorccamargo reviewed (changes requested):
<img width="1291" height="757" alt="image" src="https://github.com/user-attachments/assets/0a6ad4d7-7a50-4439-a828-132a9b39421c" />

Vou te mandar em apenas um comentário com uma lista do que ainda falta na tela e minha review atual pois vi que você apenas resolveu um dos pedidos que fiz na review anterior:
- Incluir atributos dos personagens para ficar claro além da descrição qual é o impacto daquele personagem em campo.
- As letras da lista de personagem para visualizar não estão seguindo o padrão com a fonte em pixel.
- O botão "Voltar" pode seguir o mesmo estilo do monitor de fases.
- Existe uma linha cortando a tela no meio, isso está correto ou não foi proposital? Ela não tem sentido nessa tela, retire.
- Uma das principais funcionalidades dessa tela é o jogador selecionar quais defensores ele quer para usar em fases com o máximo de 3 e miníno de 1, implemente testando tanto no modo história quanto o modo sem fim.

Murilo-Sanches commented:
Estou ciente do que preciso fazer desde a sua primeira review, pode ter certeza que quando eu acabar vou te acionar para revisar de novo.

> <img width="1291" height="757" alt="image" src="https://github.com/user-attachments/assets/0a6ad4d7-7a50-4439-a828-132a9b39421c" />
> 
> Vou te mandar em apenas um comentário com uma lista do que ainda falta na tela e minha review atual pois vi que você apenas resolveu um dos pedidos que fiz na review anterior:
> - Incluir atributos dos personagens para ficar claro além da descrição qual é o impacto daquele personagem em campo.
> - As letras da lista de personagem para visualizar não estão seguindo o padrão com a fonte em pixel.
> - O botão "Voltar" pode seguir o mesmo estilo do monitor de fases.
> - Existe uma linha cortando a tela no meio, isso está correto ou não foi proposital? Ela não tem sentido nessa tela, retire.
> - Uma das principais funcionalidades dessa tela é o jogador selecionar quais defensores ele quer para usar em fases com o máximo de 3 e miníno de 1, implemente testando tanto no modo história quanto o modo sem fim.

Murilo-Sanches committed:
refactor(wip): mudando os códigos hard-coded para apontar para a struct centralizada

Murilo-Sanches committed:
fix: evitando que o callback seja disparado duas vezes

O evento de click estava persistindo pelo frame inteiro, mas o correto é que ele seja consumido e resetado no primeiro callback.

Murilo-Sanches committed:
refactor: abrindo o leque de cores

Murilo-Sanches committed:
feat: permitindo a compra, equipar e desequipar defensores no bestiário

Também foi refatorado as fases sem fim e da história para não serem hard-coded e respeitarem o "inventário" do player.

Murilo-Sanches committed:
fix: propriedade slot é redundante e corrigindo erro de segmentation fault

O erro acontecia porque estava considerando o id do defensor como se fosse o índice dos slots, mas na verdade não é recebido o index e sim o id direto.

Murilo-Sanches committed:
style: ajustando o posicionamento dos botões

Murilo-Sanches committed:
refactor: usando a abstração do by_index

Murilo-Sanches committed:
refactor: voltando para os nomes das structs origianis já que eu removi as hard-coded das fases

Murilo-Sanches committed:
refactor: faz sentido id ser algo comum na entidade base

Murilo-Sanches committed:
style: mudando os tamanhos padrões e removendo arial

Murilo-Sanches committed:
style: mudando a posição dos botões de ação para ficar mais perto do status

Murilo-Sanches committed:
refactor: nome de váriavel melhor

Murilo-Sanches committed:
refactor: centralizando a renderização da contagem de vacinas

Murilo-Sanches committed:
style: novo background, ajustando posições e permitindo o botão ser renderizado sem conteúdo para se beneficiar apenas do callback

Murilo-Sanches committed:
refactor: escondendo a váriavel de acesso global e permitindo acesso ao valor por wrapper

Murilo-Sanches committed:
refactor: removendo o antigo immunecell

Murilo-Sanches committed:
refactor: "player" não segue o padrão snake_case e está implícito que é umaa entidade

Murilo-Sanches committed:
refactor: a representação de player não é constante então não deve ser maiúscula

Murilo-Sanches committed:
style: padronizando o último botão

paulorccamargo commented:
Aprovei para agilizar o desenvolvimento para entrar no prazo da nossa entrega, porém, o visual dessa tela precisa ser alterado com uma melhor organização. Um detalhe, muitos dos meus comentários não foram resolvidos e vamos deixar por assim pois vai levar muito tempo para serem acertados e, novamente, para o projeto ser entregue na data, vamos agilizar as tarefas ainda pendentes.

paulorccamargo commented:
Mais um detalhe que é importante evitar, você criou diversos padrões para sua PR que não foram aplicados no restante do projeto, como as váriaveis para cores e etc, evite criar novos comportamentos no projeto e se for o caso, aplique em todos os lugares.

Murilo-Sanches committed:
Merge branch 'develop' of https://github.com/guardioes-da-imunidade/guardioes-da-imunidade into 19-criar-coleção-de-defensores

Murilo-Sanches committed:
fix: na etapa de seleção do defensor não estava sendo considerado a quantidade de defensores equipados

Murilo-Sanches committed:
fix: deixei passar na resolução de conflitos

Ao selecionar o defender estava sendo considerado o índice da iteração e não o id do defensor. Também estava com o mesmo erro ao selecionar de fato o defensor onde não ficava o efeito de selecionado em verde.
